ast/compile_test.go

```diff
@@ -1502,35 +1496,70 @@ func TestCompilerCheckDuplicateImports(t *testing.T) {
 					Message:  "import must not shadow import input.foo",
 				},
 			},
-			strict: true,
-		}, {
-			note: "no strict",
+		},
+	}
+
+	runStrictnessTestCase(t, cases, true)
+}
+
+func TestCompilerCheckDeprecatedMethods(t *testing.T) {
+	cases := []strictnessTestCase{
+		{
+			note: "all() built-in",
 			module: `package test
-				import input.noconflict
-				import input.foo
-				import data.foo
-				import data.bar.foo
-				import input.bar as foo
+				p := all([true, false])
 			`,
-			strict: false,
+			expectedErrors: Errors{
+				&Error{
+					Location: NewLocation([]byte("all([true, false])"), "", 2, 10),
+					Message:  "deprecated built-in function calls in expression: all",
+				},
+			},
+		},
+		{
+			note: "any() built-in",
+			module: `package test
+				p := any([true, false])
```

Comment:
People who currently depend on `any` or `all` are likely going to resolve this by adding their own functions named like that.. so it would be good to know that works too.

```rego
package play

import future.keywords.in

any(arr) = true in arr

p := any([true, false])
```
..works?